### PR TITLE
Add Cobra and Viper

### PR DESCRIPTION
**Pull request by Adibov**
Use _Cobra_ for cli commands and _Viper_ for reading configs.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
```
Comment (emranbm) [inline cmd/root.go]: Please remove this.

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"semangit/internal/utils"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Serve semangit",
+	Long:  `With semangit you can check your versions!`, // TODO: complete this description
```
Comment (emranbm) [inline cmd/root.go]:
```suggestion
	Long:  `A simple tool to force version update in CI.`,
```

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	"semangit/internal/plugins/helm"
+	"semangit/internal/plugins/versionanalyzer"
+	"semangit/internal/utils"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "serve",
```
Comment (emranbm) [inline cmd/root.go]: serve? What's being served?  
What about `run`? or s.th. like that.

File: cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	Short: "Serve semangit",
+	Long:  `With semangit you can check your versions!`, // TODO: complete this description
+	Run: func(cmd *cobra.Command, args []string) {
+		if err := serveSemanGit(cmd, args); err != nil {
```
Comment (emranbm) [inline cmd/root.go]: *Semangit* is a single name.
```suggestion
		if err := serveSemangit(cmd, args); err != nil {
```

File: cmd/root.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+			logrus.WithError(err).Fatal("Failed to serve.")
+		}
+	},
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Register version analyzers
+	utils.PanicError(versionanalyzer.RegisterVersionAnalyzer(helm.New()))
```
Comment (emranbm) [inline cmd/root.go]: Based on our discussion, it's better to initialize explicitly in the main and keep the registration on the analyzer itself.

File: cmd/root.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+	rootCmd.Flags().StringP(
+		"repo-dir",
+		"r",
+		".",
+		"Specify repository address",
```
Comment (emranbm) [inline cmd/root.go]: Is it the *error message* or *help*? If it's a help text, seems better to have a declarative accent:  
```suggestion
		"The repository directory",
```

File: internal/config/config.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	"strings"
+)
+
+const RevisionNone = ""
+
+type Config struct {
+	RepoDir                   string                     `json:"repo_dir,omitempty"`
+	SrcRevision               string                     `json:"src_revision,omitempty"`
+	DestRevision              string                     `json:"dest_revision,omitempty"`
+	ActiveVersionAnalyzerName string                     `json:"active_version_analyzer_name,omitempty""`
```
Comment (emranbm) [inline internal/config/config.go]: As we have used **Current** in `GetCurrentVersionAnalyzerArgumentValues`, for the sake of consistency I recommend `CurrentVersionAnalyzerName` as well.

File: internal/plugins/versionanalyzer/repository.go [hunk 8]
```diff
@@ -1,14 +1,16 @@
-package versionanalyzers
+package versionanalyzer
```
Comment (emranbm) [inline internal/plugins/versionanalyzer/repository.go]: `versionAnalyzer` directory next to `base` and `helm` is confusing. I think removing the `cobra` init function will solve the cyclic import problem that probably is the root cause of this directory structure change.

**Review by emranbm — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 3 above)
Comment (Adibov) [reply cmd/root.go]: How about `analyze`?

File: cmd/serve.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
+package cmd
```
Comment (emranbm) [inline cmd/serve.go]: Please also rename this file name.

File: cmd/root.go [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+		"repo-dir",
+		"r",
+		".",
+		"The repository directory",
+	)
+	rootCmd.Flags().StringP(
+		"src-rev",
+		"s",
+		"",
+		"The source revision",
```
Comment (emranbm) [inline cmd/root.go]:
```suggestion
		"The source git revision",
```

File: cmd/root.go [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+	)
+	rootCmd.Flags().StringP(
+		"dest-rev",
+		"d",
+		"",
+		"The destination revision",
```
Comment (emranbm) [inline cmd/root.go]:
```suggestion
		"The destination git revision",
```

File: internal/versionanalyzers/config.go [hunk 12]
```diff
@@ -0,0 +1,6 @@
+package versionanalyzers
+
+type Config struct {
```
Comment (emranbm) [inline internal/versionanalyzers/config.go]: Based on our discussion, please remove `Config`

**Review by emranbm — APPROVED** [review]
Thanks!

**Pull request merged**